File: Code/new_fitervari_backend/src/main/java/at/htl/model/Exercise.java
package at.htl.model;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
public class Exercise {
    @Id
    @GeneratedValue
    private Long id;

    private String name;

    @Column(name = "creation_date")
    private LocalDateTime creationDate;

    @Column(name = "exercise_type")
    private String exerciseType;

    @Column(name = "standard_set_nr")
    private int standardSet;

    @Column(name = "official_flag")
    private boolean officialFlag;

    @ManyToOne(fetch = FetchType.EAGER)
    private Person creator;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "exercise_id")
    private List<Set> sets;

    //region GetterSetter
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDateTime creationDate) {
        this.creationDate = creationDate;
    }

    public String getExerciseType() {
        return exerciseType;
    }

    public void setExerciseType(String exerciseType) {
        this.exerciseType = exerciseType;
    }

    public int getStandardSet() {
        return standardSet;
    }

    public void setStandardSet(int standardSet) {
        this.standardSet = standardSet;
    }

    public boolean isOfficialFlag() {
        return officialFlag;
    }

    public void setOfficialFlag(boolean officialFlag) {
        this.officialFlag = officialFlag;
    }

    public Person getCreator() {
        return creator;
    }

    public void setCreator(Person creator) {
        this.creator = creator;
    }

    public List<Set> getSets() {
        return sets;
    }

    public void setSets(List<Set> sets) {
        this.sets = sets;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Exercise.kt
package at.htl_leonding.model

import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Exercise(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var name: String,
    @Column(name = "creation_date")
    var creationDate: LocalDateTime,
    @Column(name = "exercise_type")
    var exerciseType: String,
    @Column(name = "standard_set_nr")
    var standardSetNr: Int,
    @Column(name = "official_flag")
    var officialFlag: Boolean,
    @ManyToOne(cascade = [CascadeType.MERGE])
    var creator: Person,
    @OneToMany(cascade = [CascadeType.MERGE])
    @JoinColumn(name = "exercise_id")
    var sets: MutableList<Set> = mutableListOf()
) {
    fun copyValues(other: Exercise) {
        this.name = other.name
        this.creationDate = other.creationDate
        this.exerciseType = other.exerciseType
        this.standardSetNr = other.standardSetNr
        this.officialFlag = other.officialFlag
        this.creator = other.creator
        this.sets = other.sets
    }
}